src/basicfuns.jl

```diff
@@ -149,6 +149,31 @@ end
 """
 $(SIGNATURES)
 
+Return `log(abs(tanh(x)))`, carefully evaluated without intermediate calculation of `tanh(x)`.
+
+The implementation ensures `logabstanh(-x) = logabstanh(x)`.
+"""
+function logabstanh(x::Real)
+    return log1p(-2/((exp(2abs(x))+1)))
```

Comment:
As indicated by the special cases close to 0 for Float32 and Float64, maybe a safer default would be
```suggestion
    return log(abs(tanh(x)))
```

Reply:
I think the version I wrote stays more accurate overall but I'll do a test

Reply:
I don't think it's possible to do such an analysis? You can't realistically test any subtype of `Real`.

Reply:
for any `AbstractFloat` `log(tanh(x))` will be inaccurate for large x since `tanh(x)` will be very close to 1.